mockgcp/mockbigtable/cluster.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockbigtable
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	longrunning "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/pkg/storage"
+	pb "google.golang.org/genproto/googleapis/bigtable/admin/v2"
+	// pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/bigtable/admin/v2"
+)
+
+func (s *instanceAdminServer) GetCluster(ctx context.Context, req *pb.GetClusterRequest) (*pb.Cluster, error) {
+	name, err := s.parseClusterName(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.Cluster{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+func (s *instanceAdminServer) ListClusters(ctx context.Context, req *pb.ListClustersRequest) (*pb.ListClustersResponse, error) {
```

Comment:
I'm curious about this LIST call exploration. Is it for the re-reconciliation improvement or this is how the current TF-based controller uses? 
Maybe we want to have a GET method anyways. So we can add the direct controller now .

Reply:
This is just the mockgcp, so these are the calls that the terraform uses.  A bigtable Instance has a list of clusters in it: https://cloud.google.com/config-connector/docs/reference/resource-docs/bigtable/bigtableinstance

During reconciliation, terraform lists all the clusters and tries to match them.

Reply:
I see. guess TF has some special reason to do that.